### PR TITLE
Add notes for mar23

### PR DESCRIPTION
**Pull request by SriramRamesh**

### CODE REVIEW

File: mar23/sr5824.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+2. eRPC
+Popular networking software designed for niche technologies such as
+RDMA, FPGAs sacrificed generality to attain high performance. eRPC aims to achieve high performance networking without sacrificing
+generality.
+3. 1RMA
+Remote Direct Memory Access(RDMA) plays a key role in supporting 
+performance-hungry datacenter applications. 1RMA describes a new 
+approach to remote memory access suited to constraints of multi-tenant
+datacenter. 1RMA NIC is connection-free and fixed function;
+treats each RMA operatoin independently, assisting software
```
Comment (kc2853) [inline mar23/sr5824.md]: "operatoin" typo

File: mar23/sr5824.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+Remote Direct Memory Access(RDMA) plays a key role in supporting 
+performance-hungry datacenter applications. 1RMA describes a new 
+approach to remote memory access suited to constraints of multi-tenant
+datacenter. 1RMA NIC is connection-free and fixed function;
+treats each RMA operatoin independently, assisting software 
+by offering fine-grained delay measurements and fast failure 
+notifications.
+
+# Motivation
+Dataceters are growing at a massive scale and they have complex
```
Comment (kc2853) [inline mar23/sr5824.md]: "Dataceters" typo

File: mar23/sr5824.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+Dataceters are growing at a massive scale and they have complex 
+networking requirements. In case of multi-tenant datacenters,
+we have to adhere to the requirements of each customer without 
+compromising the generality of the networking software for ease 
+of usage and maintenance. This poses a variety of challenges 
+1. Network Virtualization in Multi-tenant datacenters
+Customers want to connect to the cloud datacenters with minimum 
+changes in their networking configuration and have specialized
+requirements. Multi-tenant Network Virtualization aims to satisfy
+customer requirements withotu operator intervention while
```
Comment (kc2853) [inline mar23/sr5824.md]: "withotu" typo

File: mar23/sr5824.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+	1. High Speed datacenters
+		1. Lossless fabrics for having no loss.
+     		It is a link level feature to prevent congestion-based
+     		packet drops. Ex: PFC(priority flow control)<br />
+     		PFC has the following problems:<br /> 
+     	 		1. Head-of-line blocking - performance-limiting 
+			phenomenon that occurs when a line of packets is held up by the first
+        		packet. <br />
+     			2. Deadlocks due to cyclic buffer dependencies <br />
+     			3. Complex swithc configuration <br />
```
Comment (kc2853) [inline mar23/sr5824.md]: "swithc" typo

File: mar23/sr5824.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+  		2. Large Switch buffer 
+	2. Limitations of existing options
+   		1. Exising mTCP slow for small messages(1.5 million vs 10 million
```
Comment (kc2853) [inline mar23/sr5824.md]: "Exising" typo

File: mar23/sr5824.md [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+      		64B RPC requests/s)
+	2. Recent RAMCloud has low-throughput
+      		1. Inter thread communincation for every requests
```
Comment (kc2853) [inline mar23/sr5824.md]: "communincation" typo

File: mar23/sr5824.md [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+      		2. limit system network to 1 core
+   		3. FaSST RPCs requires a lossless fabric and not handling
+      		packet loss, large messages, congestion,
+	3. Drawbacks of specilization
```
Comment (kc2853) [inline mar23/sr5824.md]: "specilization" typo

File: mar23/sr5824.md [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+where application run at massive scales, tenants require isolation and 
+security, and the workload mix changes over time. 1RMA provides 
+a new approach to adhere to the new requirements of multi-tenant 
+datacenters. 
+
+# Approaches and Trade-Offs
+
+## Network Virtualization in Multi-Tenant Datacenters
+### Introduction
+In the past fifteen years server virutalization has become dominant
```
Comment (kc2853) [inline mar23/sr5824.md]: "virutalization" typo

File: mar23/sr5824.md [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+Service Discovery - Flat L2,
+Large analytics workloads - L3,
+Some applications require multiple tiers which makes it difficult 
+for a single physical topology to support the configuration
+requirements. This results in multiple physical networks addressing
+a common topology.
+2. Address Space
+VMs get an IP from subnet of first L3 router to which they are 
+attached. This leads to a number of problems:
+1. Operators cannot move VMs to arbitary locations
```
Comment (kc2853) [inline mar23/sr5824.md]: "arbitary" typo

File: mar23/sr5824.md [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+MTDs have a set of hosts connects by physical networks.
+Host -> Multiple VMs (via hypervisor)
+VM -> Local VM/ VM in another host(via internal software switch in 
+hypervisor)
+1. Abstractions
+- Control Abstraction
+  Tenants define a set of logical network elements(or logical 
+  datapaths) that they can configure similar to physical network
+  elements. Though each tenant can have its own control plane, 
+  network hypervison provides the implementations for definied
```
Comment (kc2853) [inline mar23/sr5824.md]: "hypervison" "definied" typos

File: mar23/sr5824.md [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+4. Long running background request
+
+Contributions
+1. Common-case optimizations 66%
+2. General eRPC can
+   - Provide 2.3us median RPC latency
+   - Handle upto 10 million RPCs per second with 1 core
+   - T/f large msgs at 75 Gbps with 1 core
+   - Maintain low switch queueing during incast
+   - Perak performance with 20000 connnections per node
```
Comment (kc2853) [inline mar23/sr5824.md]: "connnections" typo

**Review by kc2853 — COMMENTED** [review]
Appreciate the notes
* Pointed out (minor) typos
* Thought it'd be nice if there was a sentence comparing the three approaches somehow

**Comment by SriramRamesh** [comment]
> Appreciate the notes
> 
> * Pointed out (minor) typos
> * Thought it'd be nice if there was a sentence comparing the three approaches somehow


Thanks for pointing out the errors. 
eRPC, net-virt, and 1RMA try to achieve high performance networking in software instead of depending on specialized hardware for general-purpose datacenters while handling node failures.

File: mar23/sr5824.md [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+
+### Trade-Offs
+eRPC is most suitable for small message size which has higher latency
+variance in specialized hardware.
+
+# Open Questions and Future Work
+1. Analysis of Incremental Server deployment for scalable NFV networks[1]
+2. Chaining of Virtual Network functions[2]
+
+# Comparison
```
Comment (jvenkit1) [inline mar23/sr5824.md]: Nice addition !

**Pull request merged**